### PR TITLE
expose circumvention settings

### PR DESCRIPTION
**Pull request by lorenzoPrimi**
See https://github.com/ooni/probe/issues/985

### CODE REVIEW

File: ooniprobe/Utility/ProxySettings.m [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+- (NSString*)getProxyString{
+    if (self.protocol == NONE)
+        return @"";
+    if (self.protocol == PSIPHON)
+        return @"psiphon://";
+    if (self.protocol == SOCKS5) {
+        // Alright, we now need to construct a new SOCKS5 URL. We are going to defer
+        // doing that to the Java standard library (er, the Android stdlib).
```
Comment (bassosimone) [inline ooniprobe/Utility/ProxySettings.m:45]: I would not mention Java here

File: ooniprobe/Utility/ProxySettings.m [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        }
+        //URI url = new URI(urlStr);
+        return urlStr;
+    }
+    return @"";
+
+}
+
++ (BOOL)isIPv6:(NSString*)hostname{
+    return false;
```
Comment (bassosimone) [inline ooniprobe/Utility/ProxySettings.m:58]: I am not very convinced by this implementation

File: ooniprobe/Utility/ProxySettings.m (hunk 1 above)
Comment (lorenzoPrimi) [reply ooniprobe/Utility/ProxySettings.m:45]: Sorry, copypasta from Android!

File: ooniprobe/Utility/ProxySettings.m (hunk 2 above)
Comment (lorenzoPrimi) [reply ooniprobe/Utility/ProxySettings.m:58]: I still don't know how to implement this, should we use a regexp?

**Pull request merged**